File: pidev/src/Entite/Centre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entite;


import java.util.ArrayList;
/**
 *
 * @author asus
 */

public class Centre {
   private int id_centre; 
    private String nom;
  
    private String description;
    private String addresse ; 
    private ArrayList<Offres> Offres;      private int id_offre ;
     private ArrayList<Offres> Produits;    private int id_produit ;
    private String categories; 

    public String getCategories() {
        return categories;
    }

    public int getId_offre() {
        return id_offre;
    }

    public void setId_offre(int id_offre) {
        this.id_offre = id_offre;
    }

    public int getId_produit() {
        return id_produit;
    }

    public void setId_produit(int id_produit) {
        this.id_produit = id_produit;
    }

    public void setCategories(String categories) {
        this.categories = categories;
    }
    

    public Centre() {
        Offres = new ArrayList<>();
         Produits = new ArrayList<>();
        
    }

    public Centre(int id_centre, String nom, String description, String addresse, ArrayList<Offres> Offres, ArrayList<Offres> Produits, String categories) {
        this.id_centre = id_centre;
        this.nom = nom;
        this.description = description;
        this.addresse = addresse;
        this.Offres = Offres;
        this.Produits = Produits;
        this.categories = categories;
    }

    
        //ajouter centre sans produits et sans offres
    public Centre(String nom, String description, String addresse, int id_offre, int id_produit, String categories) {
        this.nom = nom;
        this.description = description;
        this.addresse = addresse;
        this.id_offre = id_offre;
        this.id_produit = id_produit;
        this.categories = categories;
    }
            //ctor avec id_offre
    public Centre(String nom, String description, String addresse, int id_offre, String categories) {
        this.nom = nom;
        this.description = description;
        this.addresse = addresse;
        this.id_offre = id_offre;
        this.categories = categories;
    }
                //ctor avec id_produit
    public Centre(String nom, String description, String addresse, ArrayList<Offres> Offres, int id_produit, String categories) {
        this.nom = nom;
        this.description = description;
        this.addresse = addresse;
        this.Offres = Offres;
        this.id_produit = id_produit;
        this.categories = categories;
    }
     
      
    

    public int getId_centre() {
        return id_centre;
    }

    public String getNom() {
        return nom;
    }

    public String getDescription() {
        return description;
    }

    public String getAddresse() {
        return addresse;
    }

    public void setId_centre(int id_centre) {
        this.id_centre = id_centre;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setAddress(String addresse) {
        this.addresse = addresse;
    }

    @Override
    public String toString() {
        return "Centre{" + "id_centre=" + id_centre + ", nom=" + nom + ", description=" + description + ", addresse=" + addresse + ", Offres=" + Offres + ", Produits=" + Produits + ", categories=" + categories + '}';
    }
}
  

File: pidev1/src/AppBundle/Entity/Centre.php
<?php

namespace AppBundle\Entity;

/**
 * Centre
 */
class Centre
{
    /**
     * @var string
     */
    private $nomCentre;

    /**
     * @var string
     */
    private $adresseCentre;

    /**
     * @var integer
     */
    private $telephone;

    /**
     * @var \AppBundle\Entity\Reservation
     */
    private $idCentre;

    /**
     * @var \AppBundle\Entity\Produit
     */
    private $idProduit;

    /**
     * @var \AppBundle\Entity\Offres
     */
    private $nom;


    /**
     * Set nomCentre
     *
     * @param string $nomCentre
     *
     * @return Centre
     */
    public function setNomCentre($nomCentre)
    {
        $this->nomCentre = $nomCentre;

        return $this;
    }

    /**
     * Get nomCentre
     *
     * @return string
     */
    public function getNomCentre()
    {
        return $this->nomCentre;
    }

    /**
     * Set adresseCentre
     *
     * @param string $adresseCentre
     *
     * @return Centre
     */
    public function setAdresseCentre($adresseCentre)
    {
        $this->adresseCentre = $adresseCentre;

        return $this;
    }

    /**
     * Get adresseCentre
     *
     * @return string
     */
    public function getAdresseCentre()
    {
        return $this->adresseCentre;
    }

    /**
     * Set telephone
     *
     * @param integer $telephone
     *
     * @return Centre
     */
    public function setTelephone($telephone)
    {
        $this->telephone = $telephone;

        return $this;
    }

    /**
     * Get telephone
     *
     * @return integer
     */
    public function getTelephone()
    {
        return $this->telephone;
    }

    /**
     * Set idCentre
     *
     * @param \AppBundle\Entity\Reservation $idCentre
     *
     * @return Centre
     */
    public function setIdCentre(\AppBundle\Entity\Reservation $idCentre)
    {
        $this->idCentre = $idCentre;

        return $this;
    }

    /**
     * Get idCentre
     *
     * @return \AppBundle\Entity\Reservation
     */
    public function getIdCentre()
    {
        return $this->idCentre;
    }

    /**
     * Set idProduit
     *
     * @param \AppBundle\Entity\Produit $idProduit
     *
     * @return Centre
     */
    public function setIdProduit(\AppBundle\Entity\Produit $idProduit = null)
    {
        $this->idProduit = $idProduit;

        return $this;
    }

    /**
     * Get idProduit
     *
     * @return \AppBundle\Entity\Produit
     */
    public function getIdProduit()
    {
        return $this->idProduit;
    }

    /**
     * Set nom
     *
     * @param \AppBundle\Entity\Offres $nom
     *
     * @return Centre
     */
    public function setNom(\AppBundle\Entity\Offres $nom = null)
    {
        $this->nom = $nom;

        return $this;
    }

    /**
     * Get nom
     *
     * @return \AppBundle\Entity\Offres
     */
    public function getNom()
    {
        return $this->nom;
    }
}

